pages/[id]/bridge.tsx

```diff
@@ -44,9 +49,19 @@ const Bridge = () => {
   const { chain } = useNetwork();
   const { data: walletClient, isLoading: walletIsLoading } = useWalletClient();
 
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid },
+    watch,
+  } = useForm<FormData>({ mode: 'onChange' });
+
   // State for amount input
-  const [amount, setAmount] = useState<string>('0');
-  const rawAmount = parseUnits(amount, l1.token?.decimals || 18);
+  const amount = watch('amount', '0');
+  const rawAmount = parseUnits(
+    Boolean(errors.amount?.message) || isNaN(amount) ? '0' : amount,
```

Comment:
explicit coercion to number

```suggestion
    Boolean(errors.amount?.message) || isNaN(+amount) ? '0' : amount,
```

Reply:
https://github.com/ScopeLift/l2-flexible-voting-frontend/pull/157/commits/283e41df1bf91e1a6eba8e6be69699ab7fa38a85